### DISCUSSION TITLE
Detailed list for all holidays

### DISCUSSION THREAD

**Discussion by tossnet** [martinjw/Holiday#127] [Ideas]
When the holidays for all regions are requested, I suggest giving the result in an IEnumerable<anSpecificObject>. Currently, we can ask (for example, for Switzerland) to retrieve all public holidays, but we would need a Boolean to inform us whether day X is a public holiday or not, and whether it exists in such and such a region.

IEnumerable<nameOfAnObject> :
- datetime day
- string LocalName
- string EnglishName
- bool isPublic
- string[] Region

So what do you think?

**Comment by martinjw** [comment]
See the USAPublicHoliday which returns an object with properties
``public static Holiday NewYear(int year)``

![image](https://github.com/martinjw/Holiday/assets/2040281/78771b97-0ec6-47de-b898-5fa0547f9169)
The plan is to eventually do this for all countries, but it is more effort,

* The Holiday class has an ObservedDate as well as a HolidayDate (for eg Christmas when it is on Sunday, many countries give the holiday on the nearest day).
* There is a GetName(CultureInfo) property to get the localized name - in many countries, including CH, there are several different local languages.
* Region and IsPublic may be useful additions. 
* IdTextLocalization is a resource key. Perhaps we have an InvariantName (or EnglishName or just Name...?)
* There is a ``IList<Holiday> PublicHolidaysInformation(int year)`` on the PublicHoliday classes, which returns the Holidays.

**Comment by martinjw** [comment]
Here is a new branch with this implemented for Turkey- returning the Holiday object with dates, EnglishName and (local)Name
https://github.com/martinjw/Holiday/blob/HolidayNamesSpike/src/PublicHoliday/TurkeyPublicHoliday.cs

**Comment by tossnet** [comment]
thanks, i'll have a look
